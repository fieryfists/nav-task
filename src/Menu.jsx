import React from "react";
import Nav from "react-bootstrap/Nav";
import NavDropdown from "react-bootstrap/NavDropdown";
import { useState, useEffect } from "react";

const Menu = (props) => {
  const [menuItems, settMenuItems] = useState(props.menuItems || []);

  useEffect(() => {
    fetch("http://www.mocky.io/v2/5d3fec2b33000062009d27bc")
      .then((response) => response.json())
      .then((result) => settMenuItems(result));
  }, []);

  const handleSelect = (eventKey) => alert(`selected ${eventKey}`);

  return (
    <Nav
      style={{ position: props.position }}
      variant="pills"
      activeKey="1"
      onSelect={handleSelect}
    >
      {menuItems.map((item) => (
        <React.Fragment key={item.id}>
          {item.subItems && item.subItems.length > 0 ? (
            <NavDropdown title={item.name} id={item.id} key={item.id}>
              {item.subItems.map((subItem) => (
                <NavDropdown.Item
                  eventKey={`${item.id}.${subItem.id}`}
                  key={subItem.id}
                  id={subItem.id}
                >
                  {subItem.name}
                </NavDropdown.Item>
              ))}
            </NavDropdown>
          ) : (
            <Nav.Item>
              <Nav.Link eventKey={`${item.id}`} key={item.id} id={item.id}>
                {item.name}
              </Nav.Link>
            </Nav.Item>
          )}
        </React.Fragment>
      ))}
    </Nav>
  );
};

export default Menu;
