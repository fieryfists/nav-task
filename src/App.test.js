import React from "react";
import ReactDOM from "react-dom";
import rendered from "react-test-renderer";
import Menu from "./Menu";

describe("Menu", () => {
  const props = {
    menuItems: [
      {
        id: 0,
        name: "Item0",
        subItems: [
          {
            id: 0,
            name: "SubMenu0",
          },
          {
            id: 1,
            name: "SubMenu1",
          },
          {
            id: 2,
            name: "SubMenu2",
          },
        ],
      },
      {
        id: 1,
        name: "Item1",
        subItems: [
          {
            id: 0,
            name: "SubMenu0",
          },
          {
            id: 1,
            name: "SubMenu1",
          },
          {
            id: 2,
            name: "SubMenu2",
          },
        ],
      },
      {
        id: 2,
        name: "Item2",
        subItems: [
          {
            id: 0,
            name: "SubMenu0",
          },
          {
            id: 1,
            name: "SubMenu1",
          },
        ],
      },
      {
        id: 3,
        name: "Item3",
      },
    ],

    position: "fixed"
  };

  it("renders", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Menu {...props} />, div);
  });

  test("snapshots", () => {
    const component = rendered.create(<Menu {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
