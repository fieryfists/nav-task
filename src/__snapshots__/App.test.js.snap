// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshots 1`] = `
<div
  className="nav nav-pills"
  onKeyDown={[Function]}
  style={
    Object {
      "position": "fixed",
    }
  }
>
  <div
    className="nav-item dropdown"
  >
    <a
      aria-expanded={false}
      className="dropdown-toggle nav-link"
      data-rr-ui-event-key={null}
      id={0}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Item0
    </a>
  </div>
  <div
    className="nav-item dropdown"
  >
    <a
      aria-expanded={false}
      className="dropdown-toggle nav-link"
      data-rr-ui-event-key={null}
      id={1}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Item1
    </a>
  </div>
  <div
    className="nav-item dropdown"
  >
    <a
      aria-expanded={false}
      className="dropdown-toggle nav-link"
      data-rr-ui-event-key={null}
      id={2}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Item2
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      className="nav-link"
      data-rr-ui-event-key="3"
      id={3}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Item3
    </a>
  </div>
</div>
`;
